### PR TITLE
refactor: enable noImplicitAny table.ts 

### PR DESCRIPTION
**Pull request by laljikanjareeya**
towards #223 

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/table.ts [hunk 1]
```diff
@@ -795,16 +797,17 @@ class Table {
...
     this.database.runTransaction((err, transaction) => {
       if (err) {
         callback(err);
         return;
       }
 
-      transaction[method](this.name, rows);
-      transaction.commit(callback);
+      // tslint:disable-next-line: no-any
+      (transaction as any)![method](this.name, rows);
```
Comment (callmehiphop) [inline src/table.ts]: Why do we need to cast this to `any`?

File: src/transaction-runner.ts [hunk 2]
```diff
@@ -48,8 +48,7 @@ export interface RunTransactionOptions {
  *     already been created, and is ready to be queried and committed against.
  */
 export interface RunTransactionCallback {
-  (err: ServiceError, transaction?: null): void;
-  (err: null, transaction: Transaction): void;
+  (err: ServiceError | null, transaction?: Transaction | null): void;
```
Comment (callmehiphop) [inline src/transaction-runner.ts]: Could this just be `NormalCallback<Transaction>`?

Comment (laljikanjareeya) [reply src/transaction-runner.ts]: Done!

File: src/table.ts (hunk 1 above)
Comment (laljikanjareeya) [reply src/table.ts]: As We don't have any key-value property in `Transaction`, it gives below error:
`No index signature with a parameter of type 'string' was found on type 'Transaction'`

Comment (callmehiphop) [reply src/table.ts]: @laljikanjareeya interesting, what would happen if we changed `method` from `string` to something like

```ts
method: 'delete' | 'insert' | 'etc'
```

Comment (laljikanjareeya) [reply src/table.ts]: Yes, It's working Thanks!

**Pull request merged**